main.go

```diff
@@ -171,35 +172,21 @@ func main() {
 	// Enqueue initial request
 	updater.EnqueueRequest(&lib.Request{*urlStr})
 
-	reader := bufio.NewReader(os.Stdin)
-	running := true
-	for running {
-		cmd, err := reader.ReadString('\n')
-		if err != nil {
-			fmt.Printf("error reading from stdin: %v\n", err)
-			running = false
-			break
-		}
-
-		// Don't bother processing commands if a worker has failed and
-		// cannot be recovered
-		if workerErr != nil {
-			running = false
-			break
-		}
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
```

Comment:
Why os.Interrupt? does this catch all signals, as opposed to specify syscall.SIGINT, syscall.SIGTERM ?

Reply:
Notify requires os.Signal of which we only have Interrupt or Kill. I imagine it's a platform agnostic way of implementing it as SIGTERM doesn't exist on Windows.
https://godoc.org/os#Signal